service_container/injection_types.rst

```diff
@@ -179,6 +179,86 @@ The disadvantages of setter injection are:
 * You cannot be sure the setter will be called and so you need to add checks
   that any required dependencies are injected.
 
+Getter Injection
+----------------
+
+Another possible injection point into a class is by overriding a getter method
+to make it return the dependency::
+
+    // ...
+    abstract class NewsletterManager
+    {
+        abstract protected function getMailer(): MailerInterface;
```

Comment:
If these return types are mandatory, we must add somewhere that this only works with PHP 7. Otheriwse, we should remove them.